lib/dogma/rules/predicate_name.ex

```diff
@@ -0,0 +1,27 @@
+defmodule Dogma.Rules.PredicateName do
+  @moduledoc """
+  A rule that disallows tautological predicate names.
+  """
+
+  @behaviour Dogma.Rule
+
+  alias Dogma.Error
+
+  def test(script) do
+    script.lines
+    |> Enum.filter(&tautological_predicate/1)
+    |> Enum.map(&error/1)
+  end
+
+  defp tautological_predicate({_, line}) do
+    Regex.match?(~r{defp?\s(is|has)_\w+\?}, line)
+  end
+
+  defp error({line_num, _}) do
+    %Error{
+      rule:     __MODULE__,
+      message:  "Either `is_predicate` or `predicate?` would be sufficient",
```

Comment:
Could we have the message tell us the actual violating function name, and then recommend it with the prefix stripped?

i.e. "Favour `moduledoc?` over `has_moduledoc?`"